resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceRequestHelper.scala

```diff
@@ -153,7 +153,16 @@ private object ResourceRequestHelper extends Logging {
 
     val resInfoClass = Utils.classForName(RESOURCE_INFO_CLASS)
     val setResourceInformationMethod =
-      resource.getClass.getMethod("setResourceInformation", classOf[String], resInfoClass)
+      try {
+        resource.getClass.getMethod("setResourceInformation", classOf[String], resInfoClass)
+      } catch {
+        case e: NoSuchMethodException =>
```

Comment:
Do you really want to continue in this case? this doesn't seem like a great way to work around classpath conflict; it's just kind of ignoring it

Reply:
Good point. I can change to re-throw the exception, to make it obvious you have an issue in your class path that needs to be addressed.

Reply:
this is done.

Reply:
OK, so all this is adding is an extra message. Hm, OK, maybe worth it.
Nit: jar -> JAR, yarn -> YARN